src/lib/remark-plugins/rewrite-tutorial-links/utils/handle-learn-links.ts

```diff
@@ -26,40 +26,70 @@ export function handleTutorialLink(
 	nodePath: string,
 	tutorialMap: { [key: string]: string }
 ) {
-	const hasQueryParam = nodePath.includes('?')
-	const hasAnchorLink = nodePath.includes('#')
-	let queryParam
+	/**
+	 * Use the URL API to get each piece of the given `nodePath` that we need to
+	 * examine.
+	 */
+	const { hash, searchParams, pathname } = new URL(
+		nodePath,
+		__config.dev_dot.canonical_base_url
+	)
 
-	if (hasQueryParam) {
-		const [tutorialSlug, collectionSlug] = nodePath.split('?')
-		nodePath = tutorialSlug
-		queryParam = collectionSlug
-	}
-
-	const [, , product, filename] = nodePath.split('/') as SplitLearnPath
+	/**
+	 * Get the product slug and the tutorial's filename from the `pathname` piece
+	 * of the given `nodePath`.
+	 */
+	const [, , product, filename] = pathname.split('/') as SplitLearnPath
 
-	const tutorialSlug = [product, filename].join('/')
-	let finalSlug = tutorialMap[tutorialSlug]
+	/**
+	 * Construct the new URL's path.
+	 *   - If a collection slug is provided via the `in` query string parameter,
+	 * 	   then we build the path ourselves.
+	 *   - Otherwise, we use the path that references the default collection from
+	 *     the given `tutorialMap`.
+	 */
+	let path = ''
+	const collectionSlugParam = searchParams.get('in')
+	if (collectionSlugParam) {
+		const collectionSlug = collectionSlugParam.split('/')[1]
+		path = `/${product}/tutorials/${collectionSlug}/${filename}`
+	} else {
+		const tutorialSlug = [product, filename].join('/')
+		path = tutorialMap[tutorialSlug]
+	}
 
-	// if there is a query param, the collection name is in provided
-	// so we don't use the tutorial map
-	if (hasQueryParam) {
-		// isolate the collection name from the query
-		const [, collectionSlug] = queryParam.split('/')
-		finalSlug = `/${product}/tutorials/${collectionSlug}/${filename}`
+	/**
+	 * Construct the full string of query parameters, ignoring the `in` param that
+	 * is not needed on this platform.
+	 *
+	 * NOTE: searchParams.toString() is not used for this because it outputs
+	 * encoded characters (such as the `/` character).
```

Comment:
Question: Was this causing an issue? The docs explicitly state that the output of `toString()` is safe for use in a URL 💭 : https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/toString

Reply:
No issues, the `/` character is just more human readable than `%2F`.

Reply:
Looks like `URL.search` gives us what we want:

```
> const u = new URL('https://developer.hashicorp.com?in=collection/foo/bar&utm_param=docs')
> u.searchParams.delete('in')
> u.search
< '?utm_param=docs'
```

Reply:
To clarify here's what I'm trying to explain with `searchParams.toString()`, regarding the `/` characters:

```
> const u = new URL('https://developer.hashicorp.com?in=collection/foo/bar&param_with_slashes=this/param/has/slashes')
> u.searchParams.toString()
< 'in=collection%2Ffoo%2Fbar&param_with_slashes=this%2Fparam%2Fhas%2Fslashes'
```

Reply:
Yep, understood! The `.search` property doesn't appear to encode special characters in its output.